src/Microsoft.TestPlatform.Common/Utilities/AssemblyResolver.cs

```diff
@@ -175,9 +176,15 @@ private Assembly OnResolve(object sender, AssemblyResolveEventArgs args)
                         }
                     }
                 }
-            }
 
-            return null;
+                if (EqtTrace.IsInfoEnabled)
+                {
+                    EqtTrace.Info("AssemblyResolver: {0}: Failed to load assembly.", args.Name);
+                }
+
+                this.resolvedAssemblies[args.Name] = null;
```

Comment:
Please validate if this cache is cleaned up when extension paths are updated (IDE scenario).